### PR TITLE
Update aws-sdk-go to support pod identity

### LINKED ISSUE #4338: Upgrade AWS SDK to support EKS Pod Identity
### Describe the enhancement

EKS pod identity is an improved way to grant selected access for pods to AWS resources. Support for pod identity starts at [1.47.11 ASW SDK version](https://docs.aws.amazon.com/eks/latest/userguide/pod-id-minimum-sdk.html).  

### Describe a specific use case for the enhancement or feature

Granting elastic agent access to RDS DB for elastic agent AWS integration.  

### What is the definition of done?

Elastic agent uses AWS SDK in version higher that 1.47.11.

### PR DESCRIPTION
**Pull request by bartoszcisek**
## What does this PR do?

Update AWS SDK to version supporting EKS pod Identity 

## Why is it important?

With upgraded SDK agent can authenticate against EKS pod identity token.

## Checklist

- ~~[ ] My code follows the style guidelines of this project~~
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)

- ~~[ ] I have added an integration test or an E2E test~~

## Author's Checklist

- [ ]

## Related issues

 - Closes elastic/beats#38174

### CODE REVIEW

**Comment by pierrehilbert** [comment]
Hey @bartoszcisek could you please sign the CLA?

**Comment by bartoszcisek** [comment]
@pierrehilbert I already signed but not sure how to rerun CLA check

**Comment by bartoszcisek** [comment]
I don't have access to failed test results. Can't fix it.

File: go.mod [hunk 1]
```diff
@@ -108,6 +108,7 @@ require (
 	github.com/akavel/rsrc v0.8.0 // indirect
 	github.com/alecthomas/participle/v2 v2.1.1 // indirect
 	github.com/armon/go-radix v1.0.0 // indirect
+	github.com/aws/aws-sdk-go v1.50.26 // indirect
```
Comment (rdner) [inline go.mod:111]: why are we adding indirect dependencies manually?

The agent itself does not depend on `github.com/aws/aws-sdk-go` or `github.com/jmespath/go-jmespath`.

If we want to update indirect dependencies, we need to upgrade direct packages these dependencies are coming through.

**Pull request closed without merging**